near-sdk/src/types/gas.rs

```diff
@@ -56,6 +66,24 @@ impl From<u64> for Gas {
     }
 }
 
+fn isNum(c: char) -> bool {
+  match c {
+    '0'..='9' => true,
+    _ => false
+  }
+}
+
+impl FromStr for Gas {
+  type Err = ParseIntError;
+  fn from_str(value: &str) -> Result<Self, Self::Err> {
+    if !value.starts_with(isNum) {
+      return Err(ParseIntError{ kind: IntErrorKind::InvalidDigit })
+    }
+    let int = str::replace(value, "_", "to");
```

Comment:
```suggestion
    let int = str::replace(value, "_", "");
```
This might be your issue coming from string?